src/modules/70-pipeline/components/PipelineStudio/SavePipelinePopover/SavePipelinePopover.tsx

```diff
@@ -0,0 +1,446 @@
+  label: string
+  onClick: () => void
+  disabled?: boolean
+}
+
+export function SavePipelinePopover({
+  toPipelineStudio,
+  setGovernanceMetadata,
+  isValidYaml,
+  className = '',
+  portalClassName,
+  ...popoverProps
+}: SavePipelinePopoverProps): React.ReactElement {
+  const { isGitSyncEnabled } = React.useContext(AppStoreContext)
+  const {
+    state: { pipeline, yamlHandler, gitDetails, pipelineView, isUpdated },
+    deletePipelineCache,
+    fetchPipeline,
+    view,
+    setSchemaErrorView,
+    setSelection,
+    isReadonly,
+    updatePipelineView
+  } = usePipelineContext()
+  const [loading, setLoading] = React.useState<boolean>()
+  const { branch } = useQueryParams<GitQueryParams>()
+  const { trackEvent } = useTelemetry()
+  const { pipelineSchema } = usePipelineSchema()
+  const { showSuccess, showError, clear } = useToaster()
+  const { getString } = useStrings()
+  const { OPA_PIPELINE_GOVERNANCE } = useFeatureFlags()
+  const history = useHistory()
+  const { projectIdentifier, orgIdentifier, accountId, pipelineIdentifier, module } =
+    useParams<PipelineType<PipelinePathProps>>()
+  const isYaml = view === SelectedView.YAML
+  const [menuOpen, setMenuOpen] = React.useState(false)
+  const templatesFeatureFlagEnabled = useFeatureFlag(FeatureFlag.NG_TEMPLATES)
+  const [updatePipelineAPIResponse, setUpdatePipelineAPIResponse] = React.useState<any>()
+  const { enabled: templatesFeatureEnabled } = useFeature({
+    featureRequest: {
+      featureName: FeatureIdentifier.TEMPLATE_SERVICE
+    }
+  })
+
+  const [canEdit] = usePermission({
+    resource: {
+      resourceType: ResourceType.TEMPLATE
+    },
+    permissions: [PermissionIdentifier.EDIT_TEMPLATE]
+  })
+
+  const isTemplatesEnabled =
+    templatesFeatureEnabled &&
+    templatesFeatureFlagEnabled &&
+    canEdit &&
+    window.location.origin !== 'https://app.harness.io' &&
+    !pipeline.template
+
+  const isSaveEnabled = !isReadonly && isUpdated
+
+  const { save } = useSaveAsTemplate({ data: pipeline, type: 'Pipeline' })
+
+  const navigateToLocation = (newPipelineId: string, updatedGitDetails?: SaveToGitFormInterface): void => {
+    history.replace(
+      toPipelineStudio({
+        projectIdentifier,
+        orgIdentifier,
+        pipelineIdentifier: newPipelineId,
+        accountId,
+        module,
+        repoIdentifier: updatedGitDetails?.repoIdentifier,
+        branch: updatedGitDetails?.branch
+      })
+    )
+  }
+
+  const gotoViewWithDetails = React.useCallback(
+    ({ stageId, stepId }: { stageId?: string; stepId?: string } = {}): void => {
+      hideErrorModal()
+      // If Yaml mode, or if pipeline error - stay on yaml mode
+      if (isYaml || (!stageId && !stepId)) {
+        return
+      }
+      setSelection({ stageId, stepId })
+      updatePipelineView({
+        ...pipelineView,
+        isSplitViewOpen: true,
+        splitViewData: { type: SplitViewTypes.StageView }
+      })
+    },
+    [isYaml, pipelineView]
```

Comment:
Codacy has a fix for the issue: [React Hook React.useCallback has missing dependencies: 'hideErrorModal', 'setSelection', and 'updatePipelineView'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232314)
```suggestion
    [hideErrorModal, isYaml, pipelineView, setSelection, updatePipelineView]
```